src/utils/logger.js

```diff
@@ -0,0 +1,12 @@
+const { createLogger, transports } = require('winston');
+const { customFormatLoggerUtility } = require('./logger-custom-format');
+
+const logger = level => createLogger({
+  level,
+  transports: new transports.Console(),
+  format: customFormatLoggerUtility(),
+});
+
+exports.info = msg => logger('info').info(msg);
+exports.error = msg => logger('error').error(msg);
+exports.warn = msg => logger('warn').warn(msg);
```

Comment:
It is redundant to have the log level passed in as a string and then called as function of the logger. In your implementation you're passing in 'info' then calling the `info` method on the logger; same with **error** and **warn**. Was there any particular reason you implemented it this way? You can review the Winston documentation to see how you can use the library's API to have a cleaner implementation of this.

Reply:
Yes, there was a particular reason. If we look at line 4-8 of the same file `src/utils/logger.js`:

```
const logger = level => createLogger({
  level,
  transports: new transports.Console(),
  format: customFormatLoggerUtility(),
});
```
We can see that we are passing the type of level to the function `createLogger()`. This is required if we want to use logger.error(), logger.warn() and logger.info() respectively. I mean, if i want to use logger.error(), I need to create a new logger with level 'error' and if I want to use logger.warn(), I need to create a new logger with level 'warn". 

TLDR: In order for me to create a logger, I need to pass in the level type. I am passing 3 different levels (error, warn and info) thus creating 3 different type of loggers. Hope this was clear. I have tried other methods and this was the only one that worked for me

Reply:
There are a number of log levels in Winston. And, setting the log level in the `createLogger` function sets the **maximum** log level that will be captured when you call the logger (https://github.com/winstonjs/winston/blob/master/README.md#using-logging-levels). And, with the following log level mappings:
```
{ 
  error: 0, 
  warn: 1, 
  info: 2, 
  verbose: 3, 
  debug: 4, 
  silly: 5 
}
```
if you set the log level to `info`, then calling `logger.error`, `logger.warn` and `logger.info` will capture and print the log message. Calling `logger.verbose`, `logger.debug` or `logger.silly` will not be captured or printed.

What you are doing is:
- setting the log level to `error`, then calling `error`
- setting the log level to `warn`, then calling `warn`
- setting the log level to `info`, then calling `info`

So, this makes your implementation redundant as setting your log level to `info` when you create the logger allows you to call `error`, `warn` AND `info`.

I'll let you work out the rest of the solution.

The important bit is to dig deeper. All this included in the Winston documentation 👍

Reply:
OMG. Scratch what I said about "if i want to use logger.error(), I need to create a new logger with level 'error' and if I want to use logger.warn(), I need to create a new logger with level 'warn'". That actually was not true.... there was a reason why I thought setting the level was needed in order to use logger.error, logger.warn, etc. I tried that before (creating a logger without specifying the level) and it would always throw an error.... it turned out there was a small bug in my code with the logger.... I have updated the logger function to make it less redundant and got rid of that bug.